lib/iris/_concatenate.py

```diff
@@ -360,8 +390,63 @@ def key_func(coord):
                 coord_and_dims = _CoordAndDims(coord, tuple(dims))
                 self.aux_coords_and_dims.append(coord_and_dims)
             else:
+                metadata = _CoordMetaData(coord, None)
```

Comment:
Tiny style note:
This 'metadata = ' line is effectively the same as the call on line 388 (taken when 'if dims:' succeeds), so it might be clearer to do this once before the 'if dims:' test.